specification/k3po.control/SPEC.md

```diff
@@ -8,75 +8,58 @@ This document defines the K3po control protocol used by a test framework to comm
- command will also trigger the sending of a FINISHED event whenever the scripts finished execution on the driver. In case of failure, an
-ERROR event is sent with the cause of the failure.
+A message is either a Command or an Event.  Commands are sent by the test framework.
+Events are triggered by observed actions or state changes in the driver and are sent back to the test framework.  All messages
+are formatted as follows:
 
-- If there is no PREPARE command sent before the START command or if the PREPARE command was not successful an error with a message "Script has
- not been prepared or is still preparing" must occur. 
-
-- If the START command was already sent, an error with a message "Script has already been started" must occur.
-
-- Any time after the PREPARE command and before the DISPOSE command, the test framework can send an **AWAIT** command for a barrier existing 
-in the scripts. The command receives a response with a NOTIFIED event only when the corresponding barrier is notified. The NOTIFIED event may
- never be returned in case the barrier is never triggered. In case the barrier does not exist, an error must be returned.
+    Control Message   =   command-line LF
+                          *( header-field LF )
+                          LF
+                          [ message-body LF ]
 
-- Any time after the PREPARE command and before DISPOSE, the test framework can send a **NOTIFY** command for a barrier existing in the 
-scripts. The command notifies the barrier and a response with a NOTIFIED event is received. In case the barrier does not exist, an error must 
-be returned.
 
-- Any time after the PREPARE command and before DISPOSE, the test framework can send an **ABORT** command. The driver must abort the current 
-execution. If the execution of the scripts is already finished, no event is sent back. If the execution is aborted by this command, a FINISHED 
-event with the script that was executed so far must be returned.
+- command-line      = command LF
+- command              = token
+- header-field = field-name ":" field-value
+- LF, message-body, field-name, field-value, and token are defined in [HTTP RFC](https://tools.ietf.org/html/rfc7230).
 
-- Any time after the PREPARE command, the test framework can send a **DISPOSE** command. The driver must abort the execution and release the 
-resources in use. When done, a DISPOSED event must be sent back. At this point you can consider that the test has ended and the driver is back in the initial state.
+If the "content-length" header is specified, then a message body of the specified length must be added.
 
-- After a DISPOSE command, no other command is accepted except a new PREPARE. Any other command will have as a result an error with a message "Script has not been prepared or is still preparing". 
+Headers that are not understood / processed by a command must be ignored.
```

Comment:
The rationale for relaxed input processing is to support independent upgrade of server and client, however we always upgrade the k3po server at (say) k3po control client at the same time in the project dependencies, so it would be better to be strict on unrecognized headers.